File: textUtilsLib/src/main/java/com/shuyu/textutillib/span/ClickTopicSpan.java
package com.shuyu.textutillib.span;

import android.content.Context;
import android.view.View;

import com.shuyu.textutillib.listener.SpanTopicCallBack;
import com.shuyu.textutillib.model.TopicModel;

/**
 * 话题#点击回调
 * Created by guoshuyu on 2017/8/16.
 */

public class ClickTopicSpan extends ClickAtUserSpan {


    private TopicModel topicModel;
    private SpanTopicCallBack spanTopicCallBack;

    public ClickTopicSpan(Context context, TopicModel topicModel, int color, SpanTopicCallBack spanTopicCallBack) {
        super(context, null, color, null);
        this.topicModel = topicModel;
        this.spanTopicCallBack = spanTopicCallBack;
    }

    @Override
    public void onClick(View view) {
        super.onClick(view);
        if (spanTopicCallBack != null) {
            spanTopicCallBack.onClick(view, topicModel);
        }
    }
}

File: textUtilsLib-kotlin/src/main/java/com/shuyu/textutillib/span/ClickTopicSpan.kt
package com.shuyu.textutillib.span

import android.content.Context
import android.view.View

import com.shuyu.textutillib.listener.SpanTopicCallBack
import com.shuyu.textutillib.model.TopicModel
import com.shuyu.textutillib.model.UserModel

/**
 * 话题#点击回调
 * Created by guoshuyu on 2017/8/16.
 */

open class ClickTopicSpan(context: Context, private val topicModel: TopicModel, color: Int, private val spanTopicCallBack: SpanTopicCallBack?) : ClickAtUserSpan(context, UserModel(), color, null) {

    override fun onClick(view: View) {
        super.onClick(view)
        spanTopicCallBack?.onClick(view, topicModel)
    }
}
